pkg/gcv/result.go

```diff
@@ -34,41 +34,32 @@ const (
 
 // Result is the result of reviewing an individual resource
 type Result struct {
-	// The name of the resource as given by CAI
+	// The name of the resource as given to Config Validator
 	Name string
-	// CAIResource is the resource as given by CAI
-	CAIResource map[string]interface{}
+	// InputResource is the resource as given to Config Validator. This may be a
+	// CAI Asset or a Terraform Resource Change.
+	InputResource map[string]interface{}
 	// ReviewResource is the resource sent to Constraint Framework for review.
-	// For GCP types this is the unmodified resource from CAI, for K8S types, this is the unwrapped
-	// resource.
+	// This may be a CAI Asset, K8S resource, or Terraform Resource Change.
 	ReviewResource map[string]interface{}
 	// ConstraintViolations are the constraints that were not satisfied during review.
 	ConstraintViolations []ConstraintViolation
 }
 
 // NewResult creates a Result from the provided CF Response.
 func NewResult(
-	target string,
-	caiResource map[string]interface{},
+	target, name string,
+	inputResource map[string]interface{},
 	reviewResource map[string]interface{},
 	responses *cftypes.Responses) (*Result, error) {
 	cfResponse, found := responses.ByTarget[target]
 	if !found {
 		return nil, errors.Errorf("No response for target %s", target)
 	}
 
-	resNameIface, found := caiResource["name"]
```

Comment:
Don't we need a similar check for inputResource that it has a valid name?

Reply:
On the one hand, no - the name is now passed in directly instead of being calculated in this function. So the responsibility for validating the name would lie on the callers. Assets that are coming in via `ReviewAsset` already have this validated... but it looks like we don't currently validate the structure of assets coming in via ReviewTFResourceChange. I'll add some logic for that.